crates/iceberg-catalog/src/catalog/views/commit.rs

```diff
@@ -253,21 +254,30 @@ pub(crate) async fn commit_view<C: Catalog, A: AuthZHandler, S: SecretStore>(
         }
     }
 
-    let metadata_location =
-        storage_profile.initial_metadata_location(&view_location, Uuid::now_v7());
+    let requested_update_metadata = m.build().map_err(|e| {
+        ErrorModel::builder()
+            .code(StatusCode::BAD_REQUEST.into())
+            .message(format!("Error building metadata: {e}"))
+            .r#type("BuildMetadataError".to_string())
+            .build()
+    })?;
+
+    // TODO: Check this: this only works iff compression cannot be set to none on
```

Comment:
does this TODO still apply?

Reply:
I think so. Unfortunately, I am not familiar enough with the codebase to be able to really tell.

This might even not be the only location in code that has this problem, and if my assumptions are correct, we might have a problem every time a Table or View is changed without changing metadata compression and when metadata compression is set to the non-default value.

Assuming a Table or View is created using "none" for metadata compression, or that value was at some point set to "none".

Then a Table or View is updated, but there is no change to the metadata compression property. Each codepath that creates `CompressionCodec` from properties supplied in the request will then assume that compression codec is "gzip", as this is the default if no value for that property is supplied in `try_from_properties`.

My assumptions might be wrong, and I did not have time yet to check this thouroughly... I think I would not be able to tell if I caught every possible path in the code, or if this is a non-issue because there is some other safeguard or codepath that prevents this from happening....

I think this kinda boils down to #156 maybe not beeing a "good first issue" :wink:

Reply:
Could it be that the name of `initial_metadata_location` is a bit confusing? I'm not sure I understand the issue, the function interpolates the new location of the metadata based on view location  and some unique identifier. The initial in its name was an attempt to signal that the metadata location of a table may be different from what this function tells you, hence on reads, we need to take the metadata location from the respective db row instead of simply calling initial_metadata_location and returning that to a caller. 

It probably makes sense to rename the function to new_metadata_location. We've just recently renamed it from metadata_location to initial_metadata_location but it seems it's still confusing.

Reply:
I just re-read the code. The issue is plain and simple that I overlooked that in `commit_view`, the old metadata are loaded, the builder is then initialized from that old metadata, and the updated metadata are a result of feeding the updates from the request to the builder. I don't know how I overlooked that, but I did, and the comment is simply wrong. My bad, sorry about the confusion.